.github/workflows/ci.yml

```diff
@@ -60,17 +65,22 @@ jobs:
       DAGMC: ${{ matrix.dagmc }}
       EVENT: ${{ matrix.event }}
       VECTFIT: ${{ matrix.vectfit }}
+      LIBMESH: ${{ matrix.libmesh }}
 
     steps:
       -
         uses: actions/checkout@v2
 
-      - 
+      -
         name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-
+      -
+        name: Set up CMake
+        uses: jwlawson/actions-setup-cmake@v1.5
+        with:
+          cmake-version: 3.12.x
```

Comment:
Older versions of CMake's PkgConfig module don't seem to play nicely with libMesh's configure files, unfortunately. I'm looking into the exact version we'll need to build with libMesh so we can return an error or warning if `libmesh` is enabled and the CMake version isn't sufficient.

Reply:
Ok, I finally got to the bottom of this. The `FindPkgConfig` module shipped with CMake pre 3.12.0 has an odd quirk. If _any_ library paths are specified in the `pkgconfig` file, only those paths would be searched. So if system library locations are not included explicitly, they would not be searched when finding libraries. Here is the relevant change in CMake:

https://gitlab.kitware.com/cmake/cmake/-/commit/ac5731a7e380349f19dc319e6c31e189b5faba93

This is a problem when configuring for our libMesh build because `zlib` is present in the `libmesh.pc` file. It doesn't seem like it's standard for these files to contain system library locations, so while it's not ideal I think the best solution for now is to  require CMake >=3.12.x when libmesh is enabled. I'll adjust CI so that CMake 3.12 is only used for builds with libmesh enabled so we're still ensuring compatibility with older versions of CMake in other builds.

Reply:
Ok I was conflating issues here. I think I originally added that CMake specification to figure out what version of CMake would cause this configuration problem and isolate that problem so I could also sort out a libmesh configuration problem I was running into (which is now also resolved).

CMake versions < 3.12 when building with libmesh are still going to be problematic though, so I've added a check in the `FindLIBMESH.cmake` module to protect against this.